### DISCUSSION TITLE
Raspberry Pi Pico ADC changes the sampling rate in Free-running Sampling mode

### DISCUSSION THREAD

**Discussion by leezisheng** [micropython/micropython#15816] [RP2040 / Pico] [answered]
I am configuring the ADC sampling rate here. Due to unknown reasons, the DIV register cannot be configured here. The actual sampling period is fixed at 65.
Can anyone help me what is going wrong here？
<img width="512" alt="b078a23843d708c1ae0b4c6966c8a88" src="https://github.com/user-attachments/assets/02f1b95e-35bb-4207-a12d-32f337a84d4d">
<img width="554" alt="a91a38f1944c012e122d14c44dcbc40" src="https://github.com/user-attachments/assets/f95eda1f-2c9c-4f3c-9393-a608f5944df0">
<img width="551" alt="c39121a7385513c1579481977b0c62e" src="https://github.com/user-attachments/assets/458e9802-e07b-4459-a51d-692b5612014d">

**Comment by robert-hh** [comment, marked as answer]
The right hand expression `div_value | (int_part << 8) | frac_part` seems wrong. I would have expected it to be `(int_part << 8) | frac_part`. The previous int_part of the DIV register is 0xbfff. So if you OR 100 (0x64) to it, it will still stay 0xbfff.

**Reply by robert-hh** [reply]
Besides that, If I read back the DIV register as 0x4004c010, I get 0. And I can store and read back values.

**Reply by robert-hh** [reply]
What is the value of DMA_ADC_Transfer.DIV_REG? May it be that you are also dealing with the wrong register?

**Reply by leezisheng** [reply]
ADC_BASE = 0x4004C000
DIV_REG = ADC_BASE + 0x10

**Reply by robert-hh** [reply]
I know that this is the address of the DIV:REG. But whatr do you get if you run:

`print(hex(DMA_ADC_Transfer.DIV_REG))`

**Reply by leezisheng** [reply]
Solved,expression` (int_part << 8) | frac_part` can work . But I still have a question. According to the documentation, the div registers are all initialized to 0.
![image](https://github.com/user-attachments/assets/7bf4d764-50b7-4989-b6a5-b744ae9bd693)

**Reply by leezisheng** [reply]
![image](https://github.com/user-attachments/assets/bb5e235a-5556-4ced-bcaa-9503c4dac342)

**Reply by leezisheng** [reply]
And data bits 31:24 are reserved bits, we should not operate on them, so why my original expression does not work properly, but it does after modification.？
![image](https://github.com/user-attachments/assets/3dc41fac-4bb6-425d-ad15-e565f9133bd7)

**Reply by robert-hh** [reply]
> According to the documentation, the div registers are all initialized to 0.

Yes. That is the case. Therefore I assume that you are working with a different register controlling the DMA transfer rate.

**Reply by leezisheng** [reply]
According to the documentation, the div registers are all initialized to 0.And data bits 31:24 are reserved bits, we should not operate on them, so why my original expression does not work properly, but it does after modification.？

**Reply by leezisheng** [reply]
> I assume that you are working with a different register controlling the DMA transfer rate.
I don’t quite understand. My DIV register address setting should be correct.

**Reply by leezisheng** [reply]
I don’t quite understand. My DIV register address setting should be correct.
and Sorry, this is my first time using github issue and I am a little unfamiliar with the operations, such as citation and reference options.

**Comment by leezisheng** [comment]
And data bits 31:24 are reserved bits, we should not operate on them, so why my original expression does not work properly, but it does after modification.？
![image](https://github.com/user-attachments/assets/2d806b6c-4991-4fe4-ab51-1069e2a7f8fd)
